src/types/types.js

```diff
@@ -878,7 +878,7 @@ export type EdgeCurrencyConfig = {
   +userSettings: JsonObject | void,
 
   +changeUserSettings: (settings: JsonObject) => Promise<void>,
-  +importKey: (userInput: string) => Promise<JsonObject>
+  +importKey: (userInput: string, opts?: JsonObject) => Promise<JsonObject>
```

Comment:
We should not take the user-controlled JSON directly, but should instead take it as a named parameter:

```
+importKey: (userInput: string, opts?: { keyOptions?: JsonObject }) => Promise<JsonObject>
```

And then you would do `const { keyOptions } = opts` later on when you call the engine.